### PR TITLE
Define ResolweQuery attributes in Resolwe class

### PR DESCRIPTION
**Pull request by JureZmrzlikar**
This fix removes linter warnings of type: "Resolwe class does not
have a `data` member". The warnings are cuased since these
attributes are not defined in __init__, but in
_initialize_queries.

### CODE REVIEW

File: resdk/resources/collection.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ class BaseCollection(BaseResolweResource):
         'description', 'descriptor', 'descriptor_schema', 'settings', 'tags',
     )
 
-    ALL_PERMISSIONS = ['view', 'download', 'add', 'edit', 'share', 'owner']
+    ALL_PERMISSIONS = ['view', 'edit', 'share', 'owner']
```
Comment (mstajdohar) [inline resdk/resources/collection.py]: Could we define those on `BaseResolweResource` or make it constant?

File: resdk/resources/permissions.py [hunk 2]
```diff
@@ -234,7 +234,7 @@ def _perms_to_string(self, perms):
         perms = copy.copy(perms)
 
         # Order known permissions by importance.
-        for known_perm in ['owner', 'share', 'edit', 'add', 'download', 'view']:
+        for known_perm in ['owner', 'share', 'edit', 'view']:
```
Comment (mstajdohar) [inline resdk/resources/permissions.py]: We could abstract this bit if we made permissions a constant.

File: resdk/tests/functional/docs/e2e_docs.py [hunk 3]
```diff
@@ -70,9 +70,9 @@ def upload_reads(self, res):
             slug='upload-fastq-single',
             input={'src': os.path.join(TEST_FILES_DIR, 'reads.fastq.gz')},
         )
-        self.set_slug_and_make_public(reads, self.reads_slug, permissions=['view', 'download'])
+        self.set_slug_and_make_public(reads, self.reads_slug, permissions=['view'])
         self.set_slug_and_make_public(
-            reads.sample, self.sample_slug, permissions=['view', 'download']
+            reads.sample, self.sample_slug, permissions=['view']
```
Comment (mstajdohar) [inline resdk/tests/functional/docs/e2e_docs.py]: Can we make this one line now?

File: resdk/resources/collection.py (hunk 1 above)
Comment (JureZmrzlikar) [reply resdk/resources/collection.py]: Defined in `BaseResolweResource`.

File: resdk/resources/permissions.py (hunk 2 above)
Comment (JureZmrzlikar) [reply resdk/resources/permissions.py]: What do you mean?
`for known_perm in self.all_permissions:` (and self.all_permissions are ordered by importance)

Comment (mstajdohar) [reply resdk/resources/permissions.py]: For examples, define `ALL_PERMISSIONS` in `constants.py`.

Comment (JureZmrzlikar) [reply resdk/resources/permissions.py]: Aha, OK. Done.

**Pull request merged**